qiskit/primitives/base/base_estimator.py

```diff
@@ -254,3 +328,41 @@ def parameters(self) -> tuple[ParameterView, ...]:
             Parameters, where ``parameters[i][j]`` is the j-th parameter of the i-th circuit.
         """
         return tuple(self._parameters)
+
+
+BaseEstimator = BaseEstimatorV1
+
+
+class BaseEstimatorV2:
+    """Estimator base class version 2.
+
+    An estimator estimates expectation values for provided quantum circuit and
+    observable combinations.
+
+    An Estimator implementation must treat the :meth:`.run` method ``precision=None``
+    kwarg as using a default ``precision`` value.  The default value and methods to
+    set it can be determined by the Estimator implementor.
+    """
+
+    @staticmethod
+    def _make_data_bin(pub: EstimatorPub) -> DataBin:
+        # provide a standard way to construct estimator databins to ensure that names match
+        # across implementations
+        return make_data_bin((("evs", NDArray[np.float]), ("stds", NDArray[np.float])), pub.shape)
+
+    @abstractmethod
+    def run(self, pubs: Iterable[EstimatorPubLike], precision: float | None = None) -> Job:
```

Comment:
Why is `precision` not part of options, when everything else is?

Reply:
I also don't get the priority: if there is an option, a constructor argument, and a run argument, what has priority?

Reply:
The priority is:
 1. whatever's in a pub
 2. if that is none, whatever's in the run kwarg
 3. if that is none, the primitive chooses some value, which can be implementation specific.